### PR TITLE
Add end-to-end tests for flatdata-rs

### PR DESCRIPTION
**Pull request by VeaaC**
- Including code generation
- Small bugfix for memory storage

### CODE REVIEW

File: flatdata-rs/tests/features/build.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        if entry.path().extension().map_or(true, |x| x != "flatdata") {
+            continue;
+        }
+
+        let mut result: PathBuf = entry.path().strip_prefix(&test_cases_path).unwrap().into();
+        result.set_extension("rs");
+
+        let filename = out_path.join(&result);
+        std::fs::create_dir_all(filename.parent().unwrap()).expect("Failed to create output dir");
+        let output = std::process::Command::new("python3")
```
Comment (gferon) [inline flatdata-rs/tests/features/build.rs]: wouldn't it make sense to have all of this in a helper method of the `lib` of `flatdata-rs` so people can build their own `build.rs`?

Comment (VeaaC) [reply flatdata-rs/tests/features/build.rs]: Yes, completely agree.

File: flatdata-rs/lib/src/generator.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            output_dir
+                .join(entry.path().strip_prefix(&schemas_path).unwrap())
+                .with_extension("rs")
+        } else {
+            output_dir
+                .join(entry.path().file_name().unwrap())
+                .with_extension("rs")
+        };
+
+        println!(
```
Comment (boxdot) [inline flatdata-rs/lib/src/generator.rs]: I think you should not print inside build.rs

File: flatdata-rs/lib/src/generator.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+///
+/// # Examples
+///
+/// `build.rs`
+/// ``` ignore
+/// use std::path::PathBuf;
+///
+/// fn main() {
+///     let out_path = PathBuf::from(std::env::var("OUT_DIR").unwrap());
+///     let test_cases_path = PathBuf::from("assets/schemas");
```
Comment (boxdot) [inline flatdata-rs/lib/src/generator.rs]: We should say that only "*.flatdata" files are consider as schemas.

File: flatdata-rs/lib/src/generator.rs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+/// ``` ignore
+/// #![allow(dead_code)]
+///
+/// include!(concat!(env!("OUT_DIR"), "/path/to/my_schema.rs"));
+///
+/// // re-export if desired
+/// pub use my_schema::*;
+/// ```
+pub fn generate(output_dir: &Path, schemas_path: &Path, generator: &Path) {
+    let output_dir = output_dir.canonicalize().unwrap();
```
Comment (boxdot) [inline flatdata-rs/lib/src/generator.rs]: Many unwaps here :/

File: flatdata-rs/tests/coappearances/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "tests_coappearences"
```
Comment (boxdot) [inline flatdata-rs/tests/coappearances/Cargo.toml]: Should not this package called something with flatdata prefix?

File: flatdata-rs/lib/src/generator.rs (hunk 2 above)
Comment (VeaaC) [reply flatdata-rs/lib/src/generator.rs]: You can (only special ```cargo:*``` are doing things), but I removed cout printing (cerr will still happen from python generator itself). Input/Output are now part of error and printed in build.rs itself

File: flatdata-rs/lib/src/generator.rs (hunk 3 above)
Comment (VeaaC) [reply flatdata-rs/lib/src/generator.rs]: Done

File: flatdata-rs/lib/src/generator.rs (hunk 4 above)
Comment (VeaaC) [reply flatdata-rs/lib/src/generator.rs]: Converted to error

File: flatdata-rs/tests/coappearances/Cargo.toml (hunk 5 above)
Comment (VeaaC) [reply flatdata-rs/tests/coappearances/Cargo.toml]: Done

File: flatdata-rs/tests/coappearances/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "flatdata_tests_coappearences"
```
Comment (boxdot) [inline flatdata-rs/tests/coappearances/Cargo.toml]: Typo: coappearance

File: flatdata-rs/tests/features/src/archives/multivector.rs [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+                n::RefDataU64Index::S(s) => assert_eq!(s.x(), x as u64),
+                _ => assert!(false, "Found wrong item"),
+            }
+            assert!(iter.next().is_none(), "Too many items");
+        }
+    }
+}
+
+/*
+namespace n{
```
Comment (boxdot) [inline flatdata-rs/tests/features/src/archives/multivector.rs]: This can be removed?

Comment (VeaaC) [reply flatdata-rs/tests/features/src/archives/multivector.rs]: Done

File: flatdata-rs/tests/coappearances/Cargo.toml (hunk 6 above)
Comment (VeaaC) [reply flatdata-rs/tests/coappearances/Cargo.toml]: Done

**Pull request merged**